Start porting graphics related WebCore objects to the new IPC serialization format

gavin-apple opened the pull request:
#### d3222ec80fa20cc9e0fc3947acbecd2a75ae5c3d
<pre>
Start porting graphics related WebCore objects to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=248616">https://bugs.webkit.org/show_bug.cgi?id=248616</a>
rdar://102869400

Reviewed by Alex Christensen.

Start moving some of the graphics related serializable types over to the
new CoreIPC serialization mechanism. This includes:
    - WebCore::Gradient::LinearData
    - WebCore::Gradient::RadialData
    - WebCore::Gradient::ConicData
    - WebCore::Gradient
    - WebCore::SourceBrush::LogicalGradient
    - WebCore::Pattern::Parameters
    - WebCore::Pattern
    - WebCore::SourceBrush::Brush
    - WebCore::SourceBrush
    - WebCore::GraphicsContextGLPowerPreference
    - WebCore::GraphicsContextGLWebGLVersion
    - WebCore::GraphicsContextGLAttributes

* Source/WebCore/platform/graphics/Gradient.h:
(WebCore::Gradient::colorInterpolationMethod const):
(WebCore::Gradient::LinearData::encode const): Deleted.
(WebCore::Gradient::LinearData::decode): Deleted.
(WebCore::Gradient::RadialData::encode const): Deleted.
(WebCore::Gradient::RadialData::decode): Deleted.
(WebCore::Gradient::ConicData::encode const): Deleted.
(WebCore::Gradient::ConicData::decode): Deleted.
(WebCore::Gradient::encode const): Deleted.
(WebCore::Gradient::decode): Deleted.
* Source/WebCore/platform/graphics/GraphicsContextGLAttributes.h:
(WebCore::GraphicsContextGLAttributes::effectivePowerPreference const):
(WebCore::GraphicsContextGLAttributes::encode const): Deleted.
(WebCore::GraphicsContextGLAttributes::decode): Deleted.
* Source/WebCore/platform/graphics/Pattern.h:
(WebCore::Pattern::Parameters::encode const): Deleted.
(WebCore::Pattern::Parameters::decode): Deleted.
(WebCore::Pattern::encode const): Deleted.
(WebCore::Pattern::decode): Deleted.
* Source/WebCore/platform/graphics/SourceBrush.cpp:
(WebCore::SourceBrush::gradientSpaceTransform const):
(WebCore::SourceBrush::gradient const):
(WebCore::SourceBrush::setGradient):
* Source/WebCore/platform/graphics/SourceBrush.h:
(WebCore::operator==):
(WebCore::operator!=):
(WebCore::SourceBrush::Brush::LogicalGradient::encode const): Deleted.
(WebCore::SourceBrush::Brush::LogicalGradient::decode): Deleted.
(WebCore::SourceBrush::Brush::encode const): Deleted.
(WebCore::SourceBrush::Brush::decode): Deleted.
(WebCore::SourceBrush::encode const): Deleted.
(WebCore::SourceBrush::decode): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::FillRectWithGradient::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/257559@main">https://commits.webkit.org/257559@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6883e52dd48eda410bf9ea7760a691cc2a26f00

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99305 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8503 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32422 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108689 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168936 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103302 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9062 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85821 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91794 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106616 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105062 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6843 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90390 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33830 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21736 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76707 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2380 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23253 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2278 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/45658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5199 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8234 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42732 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3936 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.serialization.in:
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2022 Apple Inc. All rights reserved.
```
This file is unrelated.

achristensen07 commented on Source/WebCore/platform/graphics/SourceBrush.h:
```diff
@@ -33,19 +33,14 @@ namespace WebCore {
+    };
+    
     struct Brush {
-        struct LogicalGradient {
```
I don't understand why you moved this but it remains a nested struct.  I think we should leave it.

gavin-apple committed:
Start porting graphics related WebCore objects to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=248616
rdar://102869400

Reviewed by Alex Christensen.

Start moving some of the graphics related serializable types over to the
new CoreIPC serialization mechanism. This includes:
    - WebCore::Gradient::LinearData
    - WebCore::Gradient::RadialData
    - WebCore::Gradient::ConicData
    - WebCore::Gradient
    - WebCore::SourceBrush::LogicalGradient
    - WebCore::Pattern::Parameters
    - WebCore::Pattern
    - WebCore::SourceBrush::Brush
    - WebCore::SourceBrush
    - WebCore::GraphicsContextGLPowerPreference
    - WebCore::GraphicsContextGLWebGLVersion
    - WebCore::GraphicsContextGLAttributes

* Source/WebCore/platform/graphics/Gradient.h:
(WebCore::Gradient::colorInterpolationMethod const):
(WebCore::Gradient::LinearData::encode const): Deleted.
(WebCore::Gradient::LinearData::decode): Deleted.
(WebCore::Gradient::RadialData::encode const): Deleted.
(WebCore::Gradient::RadialData::decode): Deleted.
(WebCore::Gradient::ConicData::encode const): Deleted.
(WebCore::Gradient::ConicData::decode): Deleted.
(WebCore::Gradient::encode const): Deleted.
(WebCore::Gradient::decode): Deleted.
* Source/WebCore/platform/graphics/GraphicsContextGLAttributes.h:
(WebCore::GraphicsContextGLAttributes::effectivePowerPreference const):
(WebCore::GraphicsContextGLAttributes::encode const): Deleted.
(WebCore::GraphicsContextGLAttributes::decode): Deleted.
* Source/WebCore/platform/graphics/Pattern.h:
(WebCore::Pattern::Parameters::encode const): Deleted.
(WebCore::Pattern::Parameters::decode): Deleted.
(WebCore::Pattern::encode const): Deleted.
(WebCore::Pattern::decode): Deleted.
* Source/WebCore/platform/graphics/SourceBrush.cpp:
(WebCore::SourceBrush::gradientSpaceTransform const):
(WebCore::SourceBrush::gradient const):
(WebCore::SourceBrush::setGradient):
* Source/WebCore/platform/graphics/SourceBrush.h:
(WebCore::operator==):
(WebCore::operator!=):
(WebCore::SourceBrush::Brush::LogicalGradient::encode const): Deleted.
(WebCore::SourceBrush::Brush::LogicalGradient::decode): Deleted.
(WebCore::SourceBrush::Brush::encode const): Deleted.
(WebCore::SourceBrush::Brush::decode): Deleted.
(WebCore::SourceBrush::encode const): Deleted.
(WebCore::SourceBrush::decode): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::FillRectWithGradient::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/257559@main
